hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -181,10 +276,38 @@ protected void handleDeleteKeyEvent(OmKeyInfo keyInfo,
     --fileBucket[binIndex];
     nsSummary.setFileSizeBucket(fileBucket);
     nsSummaryMap.put(parentObjectId, nsSummary);
+    removeOrphanChild(keyInfo, orphanKeyMetaDataMap);
   }
 
-  protected void handleDeleteDirEvent(OmDirectoryInfo directoryInfo,
-                                      Map<Long, NSSummary> nsSummaryMap)
+  private <T extends WithParentObjectId> void removeOrphanChild(
+      T fileDirObjInfo,
+      Map<Long, OrphanKeyMetaData> orphanKeyMetaDataMap) throws IOException {
+    if (null != orphanKeyMetaDataMap) {
+      long objectID = fileDirObjInfo.getObjectID();
+      long parentObjectID = fileDirObjInfo.getParentObjectID();
+      OrphanKeyMetaData orphanKeyMetaData =
+          orphanKeyMetaDataMap.get(parentObjectID);
+      if (null == orphanKeyMetaData) {
+        orphanKeyMetaData =
+            reconNamespaceSummaryManager.getOrphanKeyMetaData(
+                parentObjectID);
+      }
+      if (null != orphanKeyMetaData) {
+        Set<Long> objectIds = orphanKeyMetaData.getObjectIds();
+        objectIds.remove(objectID);
+        if (objectIds.size() > 0) {
+          orphanKeyMetaDataMap.put(parentObjectID, orphanKeyMetaData);
+        } else {
+          orphanKeyMetaDataMap.remove(parentObjectID);
```

Comment:
removing from map is not enough, need update Db table also, May be no need remove from here, but when we commit and size is "0', than time can be removal event.

Reply:
done.